kolibri/core/logger/csv_export.py

```diff
@@ -103,7 +102,35 @@ def cache_content_title(obj):
     )
 )
 
-map_object = partial(output_mapper, labels=labels, output_mappings=mappings)
+
+def get_max_ancestor_depth():
+    max_depth = 0
+    for node in ContentNode.objects.filter(
+        content_id__in=ContentSummaryLog.objects.values_list("content_id", flat=True)
```

Comment:
We should probably batch this query, and iterate over slices of 500 content_ids at a time.

Reply:
Asking to learn: If we dont do batch operation with 500 content_ids at a time, how many content_ids at a time would it require for this function to be memory inefficient?

Reply:
I could be wrong, but my intuition here is that having a very long IN lookup will make the query not perform well, as I think it is equivalent to doing an OR-ed equal check against each value, so the more values there are, the longer the lookups will take. Possible that the query planner optimizes this, so it's worth testing to see if it makes any difference, so if you try that and see no impact, can disregard!

Reply:
Understood!
There wasn't much difference as I tested with a small data. I added iterator() still